presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -472,6 +477,11 @@ public void setOutput(Optional<Output> output)
         this.output.set(output);
     }
 
+    public void setPlan(SubPlan fragmentedPlan, PlanFlattener flattener)
+    {
+        this.flattenedPlan.set(Optional.of(flattener.flatten(fragmentedPlan, session)));
```

Comment:
I would move this to the caller `stateMachine.setPlan(flattener.flatten(fragmentedPlan, stateMachine.getSession()))`